specs/ChainSpec/LightClient.md

```diff
@@ -0,0 +1,170 @@
+# Light Client
+
+The state of the light client is defined by:
+
+1. `BlockHeaderInnerLiteView` for the current head (which contains `height`, `epoch_id`, `next_epoch_id`, `prev_state_root`, `outcome_root`, `timestamp`, the hash of the block producers set for the next epoch `next_bp_hash`, and the merkle root of all the block hashes `block_merkle_root`);
+2. The set of block producers for the current and next epochs.
+
+The `epoch_id` refers to the epoch to which the block that is the current known head belongs, and `next_epoch_id` is the epoch that will follow.
+
+Light clients operate by periodically fetching instances of `LightClientBlockView` via particular RPC end-point desdribed [below](#rpc-end-point).
+
+Light client does't need to receive `LightClientBlockView` for all the blocks. Having the `LightClientBlockView` for block `B` is sufficient to be able to verify any statement about state or outcomes in any block in the ancestry of `B` (including `B` itself). In particular, having the `LightClientBlockView` for the head is sufficient to locally verify any statement about state or outcomes in any block on the canonical chain.
+
+However, to verify the validity of a particular `LightClientBlockView`, the light client must have verified a `LightClientBlockView` for at least one block in the preceding epoch, thus to sync to the head the light client will have to fetch and verify a `LightClientBlockView` per epoch passed.
+
+## Validating Light Client Block Views
+
+```rust
+pub enum ApprovalInner {
+    Endorsement(CryptoHash),
+    Skip(BlockHeight)
+}
+
+pub struct ValidatorStakeView {
+    pub account_id: AccountId,
+    pub public_key: PublicKey,
+    pub stake: Balance,
+}
+
+pub struct BlockHeaderInnerLiteView {
+    pub height: BlockHeight,
+    pub epoch_id: CryptoHash,
+    pub next_epoch_id: CryptoHash,
+    pub prev_state_root: CryptoHash,
+    pub outcome_root: CryptoHash,
+    pub timestamp: u64,
+    pub next_bp_hash: CryptoHash,
+}
+
+pub struct LightClientBlockView {
+    pub prev_block_hash: CryptoHash,
+    pub next_block_inner_hash: CryptoHash,
+    pub inner_lite: BlockHeaderInnerLiteView,
+    pub inner_rest_hash: CryptoHash,
+    pub next_bps: Option<Vec<ValidatorStakeView>>,
+    pub approvals_after_next: Vec<Option<Signature>>,
```

Comment:
It is missing `block_merkle_root` and the corresponding spec for it.

Reply:
Yeah, the field is missing.
Regarding the spec for it, we also presently lack the spec for outcome proofs and state proofs.
I'm curious if all such info should be in the light client spec, or if we should have a separate specification for proving statements against hashes in the block.

(I don't mind including it in this spec, but if we add the spec for merkle root of blocks, we should also add all the spec for state root and outcomes root here as well I guess?)

Reply:
I think we should have a separate spec for our state and merkle trees of transactions and outcomes.

Reply:
Agree that we should have a separate spec for it.

Reply:
@nearmax @bowenwang1996 not sure if you are suggesting (1) or (2) below:

1) Have a separate spec for proofs of outcomes and state, but explain proofs of block inclusion in this spec
2) Have a separate spec for proofs of outcomes and state, and the proofs of block inclusion.

I think we should go with (2), in which case the only outstanding item for this PR is to add the missing field in this data structure.

Reply:
(2) sounds good to me